-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33746/22691

ERROR: Build errors found during clang-tidy run.
```
SimG4Core/Application/src/SteppingAction.cc:38:37: error: use of undeclared identifier 'int32' [clang-diagnostic-error]
  maxNumberOfSteps = p.getParameter<int32>("MaxNumberOfSteps");
                                    ^
SimG4Core/Application/src/SteppingAction.cc:50:36: error: use of undeclared identifier 'MaxNumberOfSteps'; did you mean 'maxNumberOfSteps'? [clang-diagnostic-error]
      << " MaxNumberOfSteps = " << MaxNumberOfSteps;
                                   ^~~~~~~~~~~~~~~~
                                   maxNumberOfSteps
--
SimG4Core/Application/src/SteppingAction.cc:106:75: error: expected ')' [clang-diagnostic-error]
  if (sAlive == tstat && edm::isNotFinite(aStep->GetTotalEnergyDeposit()) {
                                                                          ^
SimG4Core/Application/src/SteppingAction.cc:106:6: note: to match this '('
--
SimG4Core/Application/src/SteppingAction.cc:188:1: error: expected statement [clang-diagnostic-error]
}
^
Suppressed 891 warnings (890 in non-user code, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/SimG4Core/Application/interface/SteppingAction.h
+++ b/SimG4Core/Application/interface/SteppingAction.h
@@ -62,7 +62,7 @@ private:
   unsigned int numberPart;
   unsigned int ndeadRegions;
   unsigned int nWarnings;
-  unsigned int maxNumberOfSteps;
+  G4int maxNumberOfSteps;
 
   bool initialized;
   bool killBeamPipe;

--- a/SimG4Core/Application/python/NeutronBGforMuons_cff.py
+++ b/SimG4Core/Application/python/NeutronBGforMuons_cff.py
@@ -22,6 +22,7 @@ def neutronBG(process):
     process.g4SimHits.StackingAction.DeadRegions = cms.vstring()
     process.g4SimHits.StackingAction.GammaThreshold = cms.double(0.0)
     # stepping action
+    process.g4SimHits.SteppingAction.MaxNumberOfSteps = cms.int32(1000000)
     process.g4SimHits.SteppingAction.MaxTrackTime = TimeCut
     process.g4SimHits.SteppingAction.DeadRegions = cms.vstring()
     # Russian roulette disabled

--- a/SimG4Core/Application/python/g4SimHits_cfi.py
+++ b/SimG4Core/Application/python/g4SimHits_cfi.py
@@ -301,7 +301,7 @@ g4SimHits = cms.EDProducer("OscarMTProducer",
     ),
     SteppingAction = cms.PSet(
         common_maximum_time,
-        MaxNumberOfSteps        = cms.int32(10000),
+        MaxNumberOfSteps        = cms.int32(50000),
         EkinNames               = cms.vstring(),
         EkinThresholds          = cms.vdouble(),
         EkinParticles           = cms.vstring()

--- a/SimG4Core/Application/src/SteppingAction.cc
+++ b/SimG4Core/Application/src/SteppingAction.cc
@@ -35,7 +35,7 @@ SteppingAction::SteppingAction(EventAction* e, const edm::ParameterSet& p, const
   maxTrackTimes = p.getParameter<std::vector<double> >("MaxTrackTimes");
   maxTimeNames = p.getParameter<std::vector<std::string> >("MaxTimeNames");
   deadRegionNames = p.getParameter<std::vector<std::string> >("DeadRegions");
-  maxNumberOfSteps = p.getParameter<int32>("MaxNumberOfSteps");
+  maxNumberOfSteps = p.getParameter<int>("MaxNumberOfSteps");
   ekinMins = p.getParameter<std::vector<double> >("EkinThresholds");
   ekinNames = p.getParameter<std::vector<std::string> >("EkinNames");
   ekinParticles = p.getParameter<std::vector<std::string> >("EkinParticles");
@@ -47,7 +47,7 @@ SteppingAction::SteppingAction(EventAction* e, const edm::ParameterSet& p, const
       << " MaxTrackTime = " << maxTrackTime / CLHEP::ns << " ns;"
       << " MaxZCentralCMS = " << maxZCentralCMS / CLHEP::m << " m"
       << " MaxTrackTimeForward = " << maxTrackTimeForward / CLHEP::ns << " ns"
-      << " MaxNumberOfSteps = " << MaxNumberOfSteps;
+      << " MaxNumberOfSteps = " << maxNumberOfSteps;
 
   numberTimes = maxTrackTimes.size();
   if (numberTimes > 0) {
@@ -103,27 +103,25 @@ void SteppingAction::UserSteppingAction(const G4Step* aStep) {
   const G4StepPoint* postStep = aStep->GetPostStepPoint();
 
   // NaN energy deposit
-  if (sAlive == tstat && edm::isNotFinite(aStep->GetTotalEnergyDeposit()) {
+  if (sAlive == tstat && edm::isNotFinite(aStep->GetTotalEnergyDeposit())) {
     tstat = sEnergyDepNaN;
     if (nWarnings < 5) {
       ++nWarnings;
       edm::LogWarning("SimG4CoreApplication")
           << "Track #" << theTrack->GetTrackID() << " " << theTrack->GetDefinition()->GetParticleName()
-          << " E(MeV)= " << preStep->GetKineticEnergy() / MeV
-	  << " Nstep= " << theTrack->GetCurrentStepNumber()
+          << " E(MeV)= " << preStep->GetKineticEnergy() / MeV << " Nstep= " << theTrack->GetCurrentStepNumber()
           << " is killed due to edep=NaN inside PV: " << preStep->GetPhysicalVolume()->GetName() << " at "
           << theTrack->GetPosition() << " StepLen(mm)= " << aStep->GetStepLength();
     }
   }
 
   if (sAlive == tstat && theTrack->GetCurrentStepNumber() > maxNumberOfSteps) {
-    tstat = NumberOfSteps;
+    tstat = sNumberOfSteps;
     if (nWarnings < 5) {
       ++nWarnings;
       edm::LogWarning("SimG4CoreApplication")
           << "Track #" << theTrack->GetTrackID() << " " << theTrack->GetDefinition()->GetParticleName()
-          << " E(MeV)= " << preStep->GetKineticEnergy() / MeV
-	  << " Nstep= " << theTrack->GetCurrentStepNumber()
+          << " E(MeV)= " << preStep->GetKineticEnergy() / MeV << " Nstep= " << theTrack->GetCurrentStepNumber()
           << " is killed due to limit on number of steps;PV: " << preStep->GetPhysicalVolume()->GetName() << " at "
           << theTrack->GetPosition() << " StepLen(mm)= " << aStep->GetStepLength();
     }
@@ -317,8 +315,8 @@ void SteppingAction::PrintKilledTrack(const G4Track* aTrack, const TrackStatus& 
   }
 
   edm::LogVerbatim("SimG4CoreApplication")
-      << "Track #" << aTrack->GetTrackID() << " StepN= " << aTrack->GetCurrentStepNumber() 
-      << " " << aTrack->GetDefinition()->GetParticleName()
-      << " E(MeV)= " << aTrack->GetKineticEnergy() / MeV << " T(ns)= " << aTrack->GetGlobalTime() / ns
-      << " is killed due to " << typ << " inside LV: " << vname << " (" << rname << ") at " << aTrack->GetPosition();
+      << "Track #" << aTrack->GetTrackID() << " StepN= " << aTrack->GetCurrentStepNumber() << " "
+      << aTrack->GetDefinition()->GetParticleName() << " E(MeV)= " << aTrack->GetKineticEnergy() / MeV
+      << " T(ns)= " << aTrack->GetGlobalTime() / ns << " is killed due to " << typ << " inside LV: " << vname << " ("
+      << rname << ") at " << aTrack->GetPosition();
 }

